src/main/kotlin/app/revanced/patcher/PatcherContext.kt

```diff
@@ -27,35 +23,31 @@ data class PatcherContext(
          * Merge integrations.
          * @param logger A logger.
          */
-        fun merge(logger: Logger) {
-            with(context.bytecodeContext.classes) {
+        fun merge(logger: Logger, dexFileNamer: DexFileNamer) {
+            context.bytecodeContext.classes.apply {
                 for (integrations in integrations) {
-                    callback?.let { it(integrations) }
+                    logger.info("Merging $integrations")
 
-                    for (classDef in lanchon.multidexlib2.MultiDexIO.readDexFile(
-                        true,
-                        integrations,
-                        NAMER,
-                        null,
-                        null
-                    ).classes) {
+                    for (classDef in lanchon.multidexlib2.MultiDexIO.readDexFile(true, integrations, dexFileNamer, null, null).classes) {
                         val type = classDef.type
 
-                        val result = classes.findIndexed { it.type == type }
-                        if (result == null) {
+                        val existingClassIndex = this.indexOfFirst { it.type == type }
+                        if (existingClassIndex == -1) {
                             logger.trace("Merging type $type")
-                            classes.add(classDef)
+                            add(classDef)
                             continue
                         }
 
-                        val (existingClass, existingClassIndex) = result
 
                         logger.trace("Type $type exists. Adding missing methods and fields.")
 
-                        existingClass.merge(classDef, context, logger).let { mergedClass ->
-                            if (mergedClass !== existingClass) // referential equality check
-                                classes[existingClassIndex] = mergedClass
+                        get(existingClassIndex).apply {
+                            merge(classDef, context.bytecodeContext, logger).let { mergedClass ->
+                                if (mergedClass !== this) // referential equality check
+                                    set(existingClassIndex, mergedClass)
+                            }
                         }
```

Comment:
```suggestion
                        val mergedClass = merge(classDef, context.bytecodeContext, logger)
                        if (mergedClass !== get(existingClassIndex)) { // referential equality check
                            set(existingClassIndex, mergedClass)
                        }
```

Reply:
Oops, this is invalid code. Ignore this 😅

Reply:
explode